enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,358 @@
+  one based on the chosen install configuration (HA vs single node) instead of having components read/reconcile
+  the infrastructure resource.
+
+Current plan is to constrain this functionality to component level controls.
+
+2. How does the admin enable a component post-install if they change their mind about what components
+they want enabled?  Do we need/want to allow this?
+
+Turning on a component later is relatively easy (we expose a config resource for the CVO that defines
+the filter, we allow the user to remove items from the filter, the CVO will apply the previously
+filtered resources during the next reconciliation).
+
+Turning off a component later is more problematic because
+a) The CVO doesn't delete resources today, so that would be a new thing to teach it to do.
+b) Just deleting the resources for the component isn't sufficient, as the component also needs to clean
+itself up in case it created any additional resources on the cluster or contributed any configuration.
+
+Therefore we plan to support turning a component on, but not turning it off.
+
+3. What are the implications for upgrades if a future upgrade would add a component or resource which would
+have been filtered out during install time?  
+
+There should be no implication here, the CVO has the list of annotations it will filter based on, if the
+new resources match those annotations, the new resources will also be filtered(never applied to the cluster).
+
+4. How prescriptive do we want to be about what can/can't be turned off?  Components need to opt into
+this by annotating their resources, so it's not completely arbitrary.
+
+This will need to be evaluated on a case by case basis as a component considers adding the annotation
+to its resources that will allow it to be filtered out/disabled.
+
+
+5. What to do for components where disabling them has implications on other components or the way certain
+apis behave.  Example: disabling the internal registry changes the behavior of imagestreams
+(can't push to the imagestream anymore to push content to the internal registry) as well as the assumptions
+made by tools like new-app/new-build (create imagestreams that push to the internal registry).
+
+6. What to do(if anything) for components with interdependencies, to ensure a user doesn't break
+enabled components by disabling a dependency?  Options include:
+
+* Do nothing other than document dependencies so users know what not to turn off
+* Don't even annotate dependency resources for filtering, so if something is a dependency it cannot be turned off
+* Logic in the install or CVO that intelligently analyzes the filters the user has supplied and checks
+  for dependency issues (least desirable solution imho).
+
+7. Is Capabilities an ok name for these things/fields?  We'll want to get the naming right on the api to
+convey the right expectations for users.  We may revisit this name during implementation.
```

Comment:
@derekwaynecarr do we do anything equivalent to that today for other purposes?  i know the release payload has all the git commit SHAs and repo refs(I think) for all the images, right?

basically what we'd need here is to parse all the CVO-managed resource yamls to find the capabilities annotations, to get the full list of "defined" capabilities.  Certainly doesn't seem that hard for the thing that's assembling the payload image to do.

Reply:
@bparees I was imagining a dictionary mapping that let one define a capability and describe its purpose separate from its application on a particular manifest.  In this way, tooling has something meaningful to render when describing the function to put on and off.

Reply:
> I was imagining a dictionary mapping that let one define a capability and describe its purpose separate from its application on a particular manifest.

we'd want the source of the descriptions to live w/ the individual components, regardless of how they get aggregated into a single dictionary in the payload image, though.

Reply:
i've put this down as an explicit "open question" (having metadata in the payload that lists the available capabilities), not sure we need a perfect answer here, but it will probably drag ART into the implementation.  or at least the oc team to implement it in `oc adm release new`